modules/core/src/test/scala/org/tessellation/trust/SybilAttackSuite.scala

```diff
@@ -1,37 +1,134 @@
+  lazy implicit val nodeEncoder: Encoder[TrustNode] = deriveEncoder
+
+  val random: util.Random = new scala.util.Random(0L)
+
+  def benchmark(allData: List[TrustNode], filterGood: Int => Boolean): ModelBenchmark = {
+
+    val numNodes = allData.size / 2
+    val scores = SelfAvoidingWalk.runWalkFeedbackUpdateSingleNode(
+      0,
+      allData
+    )
+    val relativeLocalTrust = scores.edges.filter(e => filterGood(e.dst)).map(_.trust).sum
+    val relativeMaliciousTrust = scores.edges.filter(e => !filterGood(e.dst)).map(_.trust).sum
+
+    val eigenTrustComparisonScores = EigenTrust.calculate(allData)
+    val eigenTrustDefenseFactor = eigenTrustComparisonScores.filter(e => filterGood(e._1)).values.sum /
+      eigenTrustComparisonScores.filter(e => !filterGood(e._1)).values.sum
+
+    val defensiveFactor = relativeLocalTrust / relativeMaliciousTrust
+
+    val dattScores = DATT.calculate(DATT.convert(allData), 0)
+
+    val dattDefenseFactor = dattScores.filter(e => filterGood(e._1)).values.sum /
+      dattScores.filter(e => !filterGood(e._1)).values.sum
+
+    val modelScore = TrustModel.calculateTrust(allData, 0)
+
+    val modelDefenseFactor = modelScore.filter(e => filterGood(e._1)).values.sum /
+      modelScore.filter(e => !filterGood(e._1)).values.sum
+
+    ModelBenchmark(
+      sawDefenseFactor = defensiveFactor,
+      dattDefenseFactor = dattDefenseFactor,
+      eigenTrustDefenseFactor = eigenTrustDefenseFactor,
+      modelDefenseFactor = modelDefenseFactor
+    )
+  }
+
+  def averageBenchmark(benchmarks: List[ModelBenchmark]): ModelBenchmark =
+    ModelBenchmark(
+      benchmarks.map(_.sawDefenseFactor).sum / benchmarks.size,
+      benchmarks.map(_.dattDefenseFactor).sum / benchmarks.size,
+      benchmarks.map(_.eigenTrustDefenseFactor).sum / benchmarks.size,
+      benchmarks.map(_.modelDefenseFactor).sum / benchmarks.size
+    )
+
+  test("Data generator") {
+
+    val numNodes = 100
+    val numNetworks = 10
 
     val generator = Random.scalaUtilRandomSeedInt[IO](0).map { implicit rnd =>
       new DataGenerator[IO]
     }
     for {
-      _ <- ignore("Non-deterministic unit test!").unlessA(false)
+      _ <- ignore("Comment to modify data generation").unlessA(false)
       gen <- generator
-      data <- gen.generateData(numNodes = numNodes, edgeLogic = gen.randomPositiveEdge()).map { nodes =>
-        nodes.map { node =>
+      networkData <- List
+        .fill(numNetworks)(gen.generateData(numNodes = numNodes, edgeLogic = gen.randomPositiveEdge()))
+        .foldLeft(IO.pure(List.empty[List[TrustNode]])) {
+          case (col, next) =>
+            next.flatMap { n =>
+              col.map(_ :+ n)
+            }
+        }
+    } yield {
+      val ser = networkData.flatMap { _.map { _.asJson.noSpaces } }
+      val res = decode[TrustNode](ser.head)
+      Files
+        .write(Paths.get("./src/test/resources/trust_100x10.json"), ser.mkString("\n").getBytes(StandardCharsets.UTF_8))
```

Comment:
it's impure, shouldn't be in `yield` like that

Reply:
Changed to IO.delay in the loop -- if there's more idiomatic way let me know.